### DISCUSSION TITLE
Plot panel ability to measure time between 2 points

### DISCUSSION THREAD

**Discussion by Combinacijus** [foxglove/community#434] [Q&A] [answered]
It would be nice to have ability to measure time/period and frequency between data points. For example here I have spikes in data at seemingly regular interval. Finding time between points would be helpful for debugging

It could be like on oscilloscopes that shows time between 2 marker lines that could also be snapped to data points if wanted

![image](https://user-images.githubusercontent.com/16375702/194768803-fd95b7ee-c05b-4e13-91a2-82334e37fb1d.png)

**Comment by indirectlylit** [comment, marked as answer]
I'm happy to report that we have implemented a [point inspection and measurement tool](https://docs.foxglove.dev/docs/visualization/panels/plot#inspect-and-measure) that will help with this need.

It's available on the web now, and will be in the next desktop release. Feel free to let us know if you have feedback on the usability!

![image](https://github.com/user-attachments/assets/1e57e527-dd52-4285-9f3e-3d054a37a9bb)

**Reply by xavier-guardian** [reply]
This tool is awesome. Thank you!! Immediately so useful.
